src/main/java/gyro/azure/keyvault/VaultResource.java

```diff
@@ -0,0 +1,469 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.azure.keyvault;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.keyvault.Vault;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import gyro.azure.AzureResource;
+import gyro.azure.Copyable;
+import gyro.azure.resources.ResourceGroupResource;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+
+/**
+ * Creates a vault.
+ *
+ * **Certificate Commands**
+ *
+ * A set of certificate commands that help you ``add``, ``remove`` or ``list`` certificates from/to a specific vault that you are managing using ``gyro``. In order to use these commands make sure to add proper access policy to the vault for the service principal you are using for the ``azure-provider``.
```

Comment:
I think we should be more clear that vaults can be managed using a combination of resources defined in your Gyro configuration and commands used outside of the configuration. The first sentence has some grammatical errors.

I feel like there could be more information on using vaults. How do I get at a value stored in the vault?

Related, `package-info.java` doesn't exist for this package so documentation is not getting generated.

Reply:
Refactored the doc to give a bit more info and divided the sections properly to have the vault example first and then then the commands.

Added the package-info for the docs to be generated as well.

As for getting a value from the vault, assuming that is a certificate as that is the only one we support via this ticket, it will use the certificate finder for it, as long as you know the name of the certificate, which will give out the details of the certificate.